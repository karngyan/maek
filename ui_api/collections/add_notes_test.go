package collections_test

import (
	"context"
	"encoding/json"
	"fmt"
	"testing"
	"time"

	approvals "github.com/approvals/go-approval-tests"

	"github.com/karngyan/maek/domains/notes"

	"github.com/karngyan/maek/domains/collections"
	"github.com/karngyan/maek/ui_api/testutil"
	"github.com/stretchr/testify/assert"
)

func TestAddNotes(t *testing.T) {
	defer testutil.TruncateTables()
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	cs := testutil.NewClientStateWithUser(t)

	c, err := collections.CreateCollection(ctx, cs.Workspace.ID, cs.User.ID)
	assert.NoError(t, err)

	cbytes, err := json.Marshal(map[string]any{
		"dom": []any{},
	})

	n1, err := notes.UpsertNote(ctx, &notes.UpsertNoteRequest{
		UUID:        "uuid-1",
		Content:     cbytes,
		Created:     12345,
		Updated:     12345,
		WorkspaceID: cs.Workspace.ID,
		CreatedByID: cs.User.ID,
		UpdatedByID: cs.User.ID,
	})
	assert.NoError(t, err)

	n2, err := notes.UpsertNote(ctx, &notes.UpsertNoteRequest{
		UUID:        "uuid-2",
		Content:     cbytes,
		Created:     12345,
		Updated:     12345,
		WorkspaceID: cs.Workspace.ID,
		CreatedByID: cs.User.ID,
		UpdatedByID: cs.User.ID,
	})
	assert.NoError(t, err)

	rr, err := cs.Post(fmt.Sprintf("/v1/workspaces/%d/collections/%d/notes", c.WorkspaceID, c.ID), map[string]any{
		"noteIds": []int64{n1.ID, n2.ID},
	})
	assert.NoError(t, err)
	assert.Equal(t, 200, rr.Code)

	approvals.VerifyJSONBytes(t, rr.Body.Bytes())
}
