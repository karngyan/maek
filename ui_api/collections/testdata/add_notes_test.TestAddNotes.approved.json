{
  "collection": {
    "created": 1234567890,
    "createdById": 1,
    "description": "",
    "id": 1,
    "name": "",
    "trashed": false,
    "updated": 1234567890,
    "updatedById": 1,
    "workspaceId": 1
  },
  "notes": [
    {
      "content": {
        "dom": []
      },
      "created": 12345,
      "createdById": 1,
      "hasAudios": false,
      "hasClosedTasks": false,
      "hasCode": false,
      "hasContent": false,
      "hasFiles": false,
      "hasImages": false,
      "hasLinks": false,
      "hasOpenTasks": false,
      "hasQuotes": false,
      "hasTables": false,
      "hasVideos": false,
      "id": 1,
      "trashed": false,
      "updated": 12345,
      "updatedById": 1,
      "uuid": "uuid-1",
      "workspaceId": 1
    },
    {
      "content": {
        "dom": []
      },
      "created": 12345,
      "createdById": 1,
      "hasAudios": false,
      "hasClosedTasks": false,
      "hasCode": false,
      "hasContent": false,
      "hasFiles": false,
      "hasImages": false,
      "hasLinks": false,
      "hasOpenTasks": false,
      "hasQuotes": false,
      "hasTables": false,
      "hasVideos": false,
      "id": 2,
      "trashed": false,
      "updated": 12345,
      "updatedById": 1,
      "uuid": "uuid-2",
      "workspaceId": 1
    }
  ]
}